cookies/giles.py

```diff
@@ -533,17 +424,19 @@ def process_upload(upload_id, username):
                 upload.resource = resource
                 upload.save()
         except Exception as E:    # We f***ed something up.
+            logger.exception('Error processing Giles response')
             upload.message = str(E)
             upload.state = GilesUpload.PROCESS_ERROR
             upload.save()
             return
 
         # Depending on configuration, we may want to do things like delete local
         #  copies of files.
-        if upload.on_complete:
+        if upload.on_complete and not is_processed:
             try:
                 process_on_complete(jsonpickle.decode(upload.on_complete))
             except Exception as E:
+                logger.exception('Error executing callback')
```

Comment:
let's also log the exception

Reply:
`logger.exception` logs the traceback along with the message.